src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -2277,7 +2283,8 @@ static int getSurplusQueuedTaskCount() {
             (pool = (wt = (ForkJoinWorkerThread)t).pool) != null &&
             (q = wt.workQueue) != null) {
             int p = pool.mode & SMASK;
-            int a = p + (int)(pool.ctl >> RC_SHIFT);
+            // use max to ignore transient negative
+            int a = p + Math.max((int)(pool.ctl >> RC_SHIFT), 0);
             int n = q.top - q.base;
             return n - (a > (p >>>= 1) ? 0 :
```

Comment:
Did these condition checks miss the `==` corner-cases? i.e. the code should look like
```
            return n - (a >= (p >>>= 1) ? 0 :
                        a >= (p >>>= 1) ? 1 :
                        a >= (p >>>= 1) ? 2 :
                        a >= (p >>>= 1) ? 4 :
                        8);
```